coalib/parsing/ConfParser.py

```diff
@@ -156,7 +162,14 @@ def __parse_lines(self, lines, origin):
                                     remove_empty_iter_elements=
                                     self.__remove_empty_iter_elements),
                                     # Stop ignoring
-                            allow_appending=(keys == []))
+                            allow_appending=(
+                                keys == [] or keys == multiple_ignore_keys))
+            for key in current_section.contents:
```

Comment:
@jayvdb To take care of the test cases you suggested, I shifted the for loop outside of it's parent for loop, it solved them but it broke `test_section_inheritance` stored in `tests/settings/ConfigurationGatheringTest.py`. It's not letting me access `current_section.contents[key].value`. Can you please help?

As an alternative I tried to make the `remove` operations by making the function 
```
    def delete_newline_ignore(self,setting):
        key = setting.key
        if(key == 'ignore'):
            val = self[key]
            val.value = val._value.replace(',\n', '\n')
            val.value = val._value.replace('\n', ',')
            self.contents[key]  = val
```
in `Section.py`. It too does the job, `test_section_inheritance` throws the following error:
```
            sections['cli']['ignore'] = './user'                                                                                                                                         
            self.assertEqual(str(sections['all']['ignore']),                                                                                                                             
>                            './user, ./vendor')                                                                                                                                         
E           AssertionError: './vendor' != './user, ./vendor'                                                                                                                             
E           - ./vendor                                                                                                                                                                   
E           + ./user, ./vendor                                                                                                                                                           
                                                                                                                                                                                         
tests/settings/ConfigurationGatheringTest.py:342: AssertionError    
```
Please help

Reply:
@agrawalarpit14 is this still a problem for you? looks like you have already handled it 👍